content/blog/2022/cq-syntax.md

```diff
@@ -0,0 +1,365 @@
+---
+title: Use the Right Container Query Syntax
+sub: Size queries are stable, and shipping in browsers this summer
+author: miriam
+date: 2022-08-17
+image:
+  src: talks/container-query.jpg
+  alt: |
+    Wireframe of card elements
+    in different sized containers --
+    some laid out vertically in small spaces,
+    others horizontal when there's room.
+  position: top
+tags:
+  - Article
+  - CSS
+  - CSSWG
+  - Container Queries
+summary: |
+  Since we got a first look at
+  a Container Queries prototype
+  back in April 2021,
+  the syntax has changed a few times.
+  But now the spec is stable,
+  browsers are getting ready to ship,
+  and it's time to make sure
+  you're using the same syntax they are!
+---
+
+{% import 'embed.macros.njk' as embed %}
+
+Container Queries
+allow us to measure (or 'query')
+aspects of a 'container' element,
+and style any descendants
+based on the result of the query.
+This is very similar to
+Media Queries,
+which allow us to query the overall
+_browser viewport_.
+
+If you haven't encountered
+Container Queries before,
+there are many
+[resources available](https://css.oddbird.net/rwd/query/#articles--demos)
+on the web.
+[David Herron](/authors/davidh/)
+wrote a great
+[Quick Start Guide](/2021/04/05/containerqueries/)
+back in April 2021,
+when Chrome Canary
+shipped the first prototype of the feature.
+
+Since then
+the syntax has gone through several revisions.
+That's why we prototype --
+it allows us to learn and make changes,
+before developers are relying on the feature
+in production.
+The downside is,
+many people (including me!) have written articles,
+made demos,
+given talks,
+and released videos
+using now-out-of-date syntax.
+
+So, what syntax
+is actually shipping in browsers
+this summer?
+
+## TL;DR, a code sample
+
+```css
+main, .sidebar {
+  /* shorthand: `container: page-layout / inline-size;` */
+  container-type: inline-size;
+  container-name: page-layout;
```

Comment:
@mirisuzanne Curious, does the meaning get lost if we order these by alpha? That is also how the shorthand is ordered as well so it may make sense here too.